diff --git a/file_entity.pages.inc b/file_entity.pages.inc
index 31d0d76..b3ff55c 100644
--- a/file_entity.pages.inc
+++ b/file_entity.pages.inc
@@ -431,7 +431,7 @@ function file_entity_add_upload_submit($form, &$form_state) {
 function file_entity_upload_destination_uri(array $params, array $data = array()) {
   $params += array(
     'uri_scheme' => file_default_scheme(),
-    'file_directory' => '',
+    'file_directory' => variable_get('file_entity_default_file_directory', ''),
   );
 
   $destination = trim($params['file_directory'], '/');
